File: lib/assert.pl
#
#  simple Test::More replacement
#
use strict;

my $tests = 0;
my $errors = 0;

sub fail {
    my ($message) = @_;
    $tests++;
    $errors++;
    print STDERR "\n";
    print STDERR "#  Failed test $message\n";
}

sub ok {
    #use Data::Dumper; print Dumper(\@_);
    my ($expr, $message) = @_;
    $tests++;
    unless ($message) {
        $message = $expr;
        $expr = 0;
    }
    return if ($expr);
    $errors++;
    print STDERR "\n";
    print STDERR "#  Failed test $message\n";
}

sub is {
    my ($expected, $got, $message) = @_;
    $tests++;
    return if ($expected eq $got);
    $errors++;
    print STDERR "\n";
    print STDERR "#  Failed test $message\n";
    print STDERR "#    expected: $expected\n";
    print STDERR "#         got: $got\n";
}

sub done_testing {
    print STDERR "$. lines, $tests tests, $errors errors.\n";
    if ($errors) {
        print STDERR "FAIL\n";
        exit(2);
    }
    exit(0);
}

1;

File: lib/assert.php
<?php

#
#  simple unit testing
#
$tests = 0;
$errors = 0;

function fail($message) {
    global $tests, $errors;
    $tests++;
    $errors++;
    fwrite(STDERR, "\n");
    fwrite(STDERR, "#  Failed test $message\n");
}

function ok($expr, $message) {
    global $tests, $errors;
    $tests++;
    if (!$expr) {
        $errors++;
        fwrite(STDERR, "\n");
        fwrite(STDERR, "#  Failed test $message\n");
    }
}

function is($got, $expected, $message) {
    global $tests, $errors;
    $tests++;
    if ($expected == $got) {
        return;
    }
    $errors++;
    fwrite(STDERR, "\n");
    fwrite(STDERR, "#  Failed test $message\n");
    fwrite(STDERR, "#         got: $got\n");
    fwrite(STDERR, "#    expected: $expected\n");
}

function plural($count, $word) {
    return "$count $word" . ($count == 1 ? " " : "s ");
}

function done_testing() {
    global $tests, $errors;
    fwrite(STDERR, plural($tests, "test") . plural($errors, "error") . "\n");
    if ($errors) {
        fwrite(STDERR, "FAIL\n");
        exit(2);
    }
    fwrite(STDERR, "PASS\n");
    exit(0);
}

?>
